test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -109,6 +111,35 @@ class SyncConfigTests {
         assertNotNull(config.errorHandler)
     }
 
+    @Test
+    fun compactOnLaunch_default() {
+        val user = createTestUser()
+        val config = SyncConfiguration.Builder(
+            schema = setOf(ParentPk::class, ChildPk::class),
+            user = user,
+            partitionValue = DEFAULT_PARTITION_VALUE
+        ).build()
+        assertNull(config.compactOnLaunchCallback)
+    }
+
+    @Test
```

Comment:
Maybe also add a test that it is set to `DEFAULT_COMPACT_ON_LAUNCH_CALLBACK` if called without an argument.